test/cypress/integration/simplified-account/service-settings/card-payments/updates.cy.js

```diff
@@ -102,34 +112,157 @@ describe('Card payment updates', () => {
+      patchAccountByServiceExternalIdAndAccountTypeUpdateGooglePaySuccess(SERVICE_EXTERNAL_ID, ACCOUNT_TYPE, true)
     ])
     cy.visit(baseUrl + '/google-pay')
     cy.get('h1').should('contain.text', 'Google Pay')
-    cy.get('.service-settings-nav__li--active').within(() => {
-      cy.get('#card-payments').should('contain.text', 'Card payments')
+    checkSettingsNavigation('Card payments', baseUrl)
+    cy.get('input#google-pay-on').click()
+    cy.contains('button', 'Save changes').click()
+    cy.get('.govuk-heading-l').should('contain.text', 'Card payments')
+  })
+
+  it('should allow update of Google Pay with active Worldpay credential', () => {
+    const googlePayMerchantId = '0123456789abcde'
+    const googlePayMerchantIdError = 'Enter a Google Pay merchant ID'
+    setupStubs({
+      allowGooglePay: true,
+      gatewayAccountPaymentProvider: WORLDPAY,
+      gatewayAccountCredentials: [
+        WORLDPAY_CREDENTIAL_IN_ACTIVE_STATE
+      ]
     })
+    cy.task('setupStubs', [
+      patchAccountByServiceExternalIdAndAccountTypeUpdateGooglePaySuccess(SERVICE_EXTERNAL_ID, ACCOUNT_TYPE, true),
+      patchAccountByServiceExternalIdAndAccountTypeUpdateGooglePayMerchantIdSuccess(
+        SERVICE_EXTERNAL_ID,
+        ACCOUNT_TYPE,
+        WORLDPAY_CREDENTIAL_IN_ACTIVE_STATE.external_id,
+        {
+          googlePayMerchantId,
+          userExternalId: USER_EXTERNAL_ID
+        })
+    ])
+    cy.visit(baseUrl + '/google-pay')
+    cy.get('h1').should('contain.text', 'Google Pay')
+    checkSettingsNavigation('Card payments', baseUrl)
+    cy.get('.govuk-error-summary').should('not.exist')
     cy.get('input#google-pay-on').click()
     cy.contains('button', 'Save changes').click()
+    cy.get('.govuk-error-summary')
+      .should('exist')
+      .should('contain', googlePayMerchantIdError)
+    cy.get('input#google-pay-merchant-id')
+      .should('have.class', 'govuk-input--error')
+    cy.get('#google-pay-merchant-id-error')
+      .should('contain.text', googlePayMerchantIdError)
+    cy.get('input#google-pay-merchant-id').click()
+      .clear({ force: true })
+      .type(googlePayMerchantId)
+    cy.contains('button', 'Save changes').click()
     cy.get('.govuk-heading-l').should('contain.text', 'Card payments')
   })
+
+  it('should not allow update of Google Pay with inactive Worldpay credential', () => {
+    setupStubs({
+      allowGooglePay: false,
+      gatewayAccountPaymentProvider: WORLDPAY,
+      gatewayAccountCredentials: [
+        WORLDPAY_CREDENTIAL_IN_CREATED_STATE
+      ]
+    })
+    cy.visit(baseUrl + '/google-pay')
+    cy.get('h1').should('contain.text', 'Card payments')
+    checkSettingsNavigation('Card payments', baseUrl)
+    cy.contains('.govuk-summary-list__key', 'Google Pay')
+      .parent()
+      .within(() => {
+        cy.contains('Off (complete Worldpay details to switch on)').should('exist')
+        cy.contains('a', 'Change').should('not.exist')
```

Comment:
out of interest is it still possible to visit this URL? perhaps we should have a cypress test that it can't be visited/POSTed to

Reply:
this test is doing exactly that, visting the google pay setting url and validating that you are redirected back to the Card payments settings

Reply:
oh yeah i see, I missed that it's visiting `/google-pay` - this doesn't cover off the possibility of POSTing a form to that URL, although maybe testing that is overkill

Reply:
added a POST test anyway 🤷🏻‍♂️